### PR TITLE
Week 5 assessment jacksonpruett

### PR DESCRIPTION
**Pull request by JacksonPruett**
# Submitting Week 5 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X] Career Readiness
- [X] Coding with AI
- [X] Interview Questions
- [X] JavaScript Challenges
- [X] Ruby Challenges

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

I'll be honest I'm sure theres an easier way to complete the JS challenges. But after I had gone through the syllabus lessons 3-4 times and it still wasn't clicking I turned to w3schools and stack overflow until I stopped getting errors. I have since gone back and learned about what's being used in my functions so I can use it again in the future and explain if necessary. BUT if you know simpler ways please point me in the right direction lol because that took an enormous amount of research and trial and error. What was probably meant to take a couple hours took me half a day.

That and any other guidance you see fit to offer would be greatly appreciated! Hope you have/had an awesome weekend!

### CODE REVIEW

**Comment by JacksonPruett** [comment]
Also worth noting I did have help from chatGPT (I raised my hand this morning lol) on the full house one and then I had assistance from the same with getting the syntax correctly written on the one before it. But I have since studied what was used and understand it. I am confident I could use the methods I learned without help from AI in the future. JS is hard.

File: coding-with-ai.rb [hunk 1]
```diff
@@ -6,11 +6,20 @@
...
   result
 end
 
-p totals
\ No newline at end of file
+# print total vote counts for each taco in the array
+p totals
+# {"fish taco"=>9, "california burrito"=>4, "tacos al pastor"=>3, "carnitas tacos"=>4, "tacos de barbacoa"=>2}
+
+# Prompt: Create the code that takes in the array, iterates over it, and returns a hash with each taco type and the total number of times it was mentioned or "voted" in the array.
```
Comment (triciasykes) [inline coding-with-ai.rb:25]: Well done!

File: interview-questions.md [hunk 2]
```diff
@@ -8,32 +8,45 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 Research and define the following terms to the best of your ability.
 
-1. Model validations:
+1. Model validations: After training a model, we perform model validation by checking how well it works with a separate set of testing data. This testing data can be part of the same dataset used for training or a different one.
 
-2. Params:
+2. Params: URL parameters, also referred to as params or query strings, serve as a means to include extra information in a controller method. This information can be used to query the database or dynamically alter the view.
 
-3. API:
+3. API: An application programming interface is a way for two or more computer programs to communicate.
```
Comment (triciasykes) [inline interview-questions.md:52]: Love your work here!  Go Jackson!

File: javascript-challenges.test.js [hunk 3]
```diff
@@ -14,30 +14,98 @@
...
+        frequency[number] = (frequency[number] || 0) + 1 // Personal note: number parameter is in square brackets b/c its a number
+    })
+    // since arrays are objects, I can use the Object method.
+    // values is equal to the values of the Object, frequency.
+    const values = Object.values(frequency)
+    // return the included pair (2) values AND the included 3 of a kind (3) values
+    // if both conditions are true it will return true, otherwise it will return false.
+    return values.includes(2) && values.includes(3)
+}
+
```
Comment (triciasykes) [inline javascript-challenges.test.js:111]: Jackson, based on your request...I think either you're overthinking it, not recognizing that you know more than you do, or maybe not breaking down the problems far enough.  Probably a combination of all 3 😁.  There's no need for regex or anything like that in the first one. Although not as sexy, you can use a conditional statement and reassign the letters when appropriate. The second one, think about what the first thing you do when you get your hand when playing cards.  Then think about the win conditions.  Again, conditional will take care of it.  More code isn't always a bad thing.  Don't get hung up on the oneliners and such.  I've seen you make so much progress.  You can do this!

File: ruby-challenges.rb [hunk 4]
```diff
@@ -4,22 +4,74 @@
...
+my_bike.pedal_faster(18)
+# "My bike has 2 Wheels and is going 28 mph."
+my_bike.brake(5)
+# "My bike has 2 Wheels and is going 23 mph."
+my_bike.brake(25)
+# "My bike has 2 Wheels and is going 0 mph."
+p my_bike.bike_info
+
+
+
```
Comment (triciasykes) [inline ruby-challenges.rb:77]: great job!

**Review by triciasykes — APPROVED** [review]
Jackson, you're doing great.  I left some comments for you based on your request.  I think doing more problems, getting that experience will really elevate you.  This shows Advanced Beginner level.  Keep going!

**Pull request merged**